src/components/Notification/Notification.tsx

```diff
@@ -28,6 +28,7 @@ export const DefaultTitles = {
 type NotificationAction = {
   label: string;
   onClick: () => void;
+  disabled?: boolean;
 };
```

Comment:
Another option would be to allow passing any button props to these actions to make them more flexible.

```suggestion
type NotificationAction = {
  label: string;
  onClick: () => void;
} & Partial<ButtonProps>;
```

Reply:
This was actually my initial plan, to replace the underlying button with an **ActionButton** and extend the Props. For my use case I needed access to both the `disabled` and `loading` props on an **ActionButton**. But David said that he wasn't sure about using an **ActionButton**. I can extend the **ButtonProps** and leave the underlying **Button** as is instead of using an **ActionButton**.

![image](https://github.com/user-attachments/assets/f0f742db-ba80-443b-b2cc-94c6e6bdebd3)
_Note: onClick would just be removed from the **Omit** group and from the **NotificationAction** itself since Button already has the prop. And we could wrap the Omit with a Partial_

What would you suggest?


Reference: https://chat.canonical.com/canonical/pl/ax8fwwcrz3fdupjaag6gt4fimh

Reply:
Another option would be to allow `ReactNode` as well as `NotificationAction` so that you can pass an `ActionButton` or any other component:

```
actions?: (NotificationAction | ReactNode)[];
```

And then you'd also need to update where it does `actions.map` to only map to a button when it's a `NotificationAction`.

Reply:
That would be even better! Do we need an array of `ReactNode` though? Can we just do this and avoid mapping altogether if it's a ReactNode?

```ts
actions?: NotificationAction[] | ReactNode;
```

Also, I believe we would still keep the `type="button"` change and the `disabled` prop can be removed

Reply:
> That would be even better! Do we need an array of `ReactNode` though? Can we just do this and avoid mapping altogether if it's a ReactNode?
> 
> ```ts
> actions?: NotificationAction[] | ReactNode;
> ```

True, though the benefit of allowing a mix will be that if you only want to use ReactNode for some actions and standard NotificationActions for others you won't have to pass in a button that matches what the component defines with the correct classes etc (for the NotificationActions).

> 
> Also, I believe we would still keep the `type="button"` change and the `disabled` prop can be removed

Yep that sounds good.

Reply:
Ah, fair point. I'll make the changes then. Thank you!

Reply:
Done